packages/runtime/container-runtime/src/connectionTelemetry.ts

```diff
@@ -101,43 +116,43 @@ class OpPerfTelemetry {
 		});
 		this.deltaManager.on("disconnect", () => {
 			this.sequenceNumberForMsnTracking = undefined;
-			this.clientSequenceNumberForLatencyStatistics = undefined;
-			this.opProcessingTimes = {};
-			this.opPerfData = {};
 			this.connectionOpSeqNumber = undefined;
 			this.firstConnection = false;
-			this.pongCount = 0;
+			this.latencyStatistics.clear();
 		});
 
 		this.deltaManager.outbound.on("push", (messages) => {
 			for (const msg of messages) {
 				if (
 					msg.type === MessageType.Operation &&
-					this.clientSequenceNumberForLatencyStatistics === msg.clientSequenceNumber
+					msg.clientSequenceNumber % OpPerfTelemetry.OP_LATENCY_SAMPLE_RATE === 0
```

Comment:
This change can subtly break things... it assumes that `msg.clientSequenceNumber` is always monotonically increasing and has no gaps. That might be correct today but I wouldn't want to tie this functionality to that assumption.

It also breaks the case of unsampled telemetry, because we'd only be capturing stats for one out of every `OP_LATENCY_SAMPLE_RATE` messages. For unsampled telemetry to work, I think we just eat the need to calculate stats for every message. In order to avoid that when we are applying sampling, we'd need to reach into the sampler in these checks and that just doesn't feel right.

Reply:
Undid this change.